### PR TITLE
gpm: remove ncurses null override

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/gpm/default.nix [hunk 1]
```diff
@@ -43,7 +43,7 @@ stdenv.mkDerivation rec {
   configureFlags = [
     "--sysconfdir=/etc"
     "--localstatedir=/var"
-    (if ncurses == null then "--without-curses" else "--with-curses")
+    (if withNcurses then "--without-curses" else "--with-curses")
```
Comment (smancill) [inline pkgs/servers/gpm/default.nix:46]:
```suggestion
    (if withNcurses then "--with-curses" else "--without-curses")
```

**Comment by trofi** [comment]
4d47a0ef9003534c48f6275f3b31b2ca38720599 `gpm: remove ncurses null override (#183810)` somehow broke `w3m` in `staging`:

```
$ nix build -f. w3m -L
...
w3m> gcc  -I. -I. -g -O2 -I./libwc -I/nix/store/nzxhqf6dzhqwy72sg0b2qyvscr4i0cdv-openssl-1.1.1q-dev/include  -DHAVE_CONFIG_H -DAUXBIN_DIR=\"/nix/store/1rwpkdy595ls0882d6qii0ynz7563fid-w3m-0.5.3+git20190105/libexec/w3m\" -DCGIBIN_DIR=\"/nix/store/1rwpkdy595ls0882d6qii0ynz7563fid-w3m-0.5.3+git20190105/libexec/w3m/cgi-bin\" -DHELP_DIR=\"/nix/store/1rwpkdy595ls0882d6qii0ynz7563fid-w3m-0.5.3+git20190105/share/w3m\" -DETC_DIR=\"/nix/store/1rwpkdy595ls0882d6qii0ynz7563fid-w3m-0.5.3+git20190105/etc\" -DCONF_DIR=\"/nix/store/1rwpkdy595ls0882d6qii0ynz7563fid-w3m-0.5.3+git20190105/etc/w3m\" -DRC_DIR=\"~/.w3m\" -DLOCALEDIR=\"/nix/store/1rwpkdy595ls0882d6qii0ynz7563fid-w3m-0.5.3+git20190105/share/locale\" -o w3m main.o file.o buffer.o display.o etc.o search.o linein.o table.o local.o form.o map.o frame.o rc.o menu.o mailcap.o image.o symbol.o entity.o terms.o url.o ftp.o mimehead.o regex.o news.o func.o cookie.o history.o backend.o keybind.o anchor.o parsetagx.o tagtable.o istream.o version.o  -lm -lX11 -L. -lindep  -lgc -L./libwc -lwc -L/nix/store/sxgs416j8z3x9mj97939syyp8cv6064b-openssl-1.1.1q/lib -lssl -lcrypto -lssl -lcrypto -lgpm -ltinfo
w3m> /nix/store/dpxy5cr7cm0nch0qvq2qygs9jwnmhiy5-binutils-2.38/bin/ld: terms.o: in function `do_getch':
w3m> /build/source/terms.c:2174: undefined reference to `Gpm_Wgetch'
```

**Comment by trofi** [comment]
Ah, it's because `w3m` uses `gpm-ncurses = gpm.override { inherit ncurses; };` and `gpm` now uses `withNcurses = false;`.

**Comment by trofi** [comment]
Proposed possible fix as https://github.com/NixOS/nixpkgs/pull/184146

**Pull request merged**